packages/Search/src/DTK_FineSearchFunctor.hpp

```diff
@@ -0,0 +1,361 @@
+    {
+        return Intrepid2::Impl::Triangle<3>::checkPointInclusion(
+            reference_point, threshold );
+    }
+};
+
+class Triangle_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_TRI_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Triangle<6>::checkPointInclusion(
+            reference_point, threshold );
+    }
+};
+
+class Wedge_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C1_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<6>::checkPointInclusion( reference_point,
+                                                               threshold );
+    }
+};
+
+class Wedge_18
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<18>::checkPointInclusion( reference_point,
+                                                                threshold );
+    }
+};
+}
+
+namespace Functor
+{
+template <typename CellType, typename DeviceType>
```

Comment:
No, it's not to extend it. It's to pack 2 parameters into one. We need `shards::CellTopology` to use `checkPointInclusion` and we need another template on the basis function to use `mapToReferenceFrame`. Given a the topology, I pick the basis function.